contrib/doc/GafferUserGuide/tutorials/configurationFilesExample.txt

```diff
@@ -7,14 +7,14 @@ Gaffer applications are intended to be easily extensible and customisable, and t
 ==== Startup file locations
 The location of Gaffer's configuration files are specified using the `GAFFER_STARTUP_PATHS` environment variable. This is a colon separated list of paths to directories where the startup files reside. Config directories at the end of the list are executed first, allowing them to be overridden by config directories earlier in the list.
 
-Gaffer automatically adds the `~/gaffer/startup` config directory to the `GAFFER_STARTUP_PATHS` to allow users to create their own config files without having to faff around with the environment. This user level config is run last, allowing it to take precedence over all other configuration files.
+Gaffer automatically adds the `~/gaffer/<gafferCompatibilityVersion>/startup` config directory to the `GAFFER_STARTUP_PATHS` to allow users to create their own config files without having to faff around with the environment. This user level config is run last, allowing it to take precedence over all other configuration files.
```

Comment:
This troubles me because I want users to have a single location where they can stuff ad-hoc configs without having to think about versioning or anything else. I don't want them to be in a situation where they're using two or more Gaffer versions and therefore having to maintain multiple sets of configs.

One solution might be to keep `~/gaffer/startup` in the paths, running it before `~/gaffer/startup<gafferCompatibilityVersion>`. But then the initial "copy the unversioned config" step needs to only take the files that have been auto-saved by Gaffer, and not the user's own files. This makes me wonder if we should even be sharing the same directory for user-generated files and gaffer-generated ones - there's a danger of a clash when Gaffer wants to introduce a new file in the future. Given the frequency of version changes, I think I'm also going to be bothered by the number of version folders piling up inside `~/gaffer`. And possibly also the fact that preferences I change in one version aren't going to be available in the previous one. Some of this could be alleviated by moving just Gaffer's auto-generated files to something like `~/gaffer/preferences/<gafferCompatibilityVersion>`, but I am beginning to wonder if the cure is worse than the disease...

Reply:
I like the idea of separating the auto generated stuff from the user maintained stuff. Shall I go with startup and preferences folders then?

Reply:
> I like the idea of separating the auto generated stuff from the user maintained stuff. Shall I go with startup and preferences folders then?

I think it's either that or drop this completely. What did you think to the worry about it being annoying to save preferences in Gaffer N+1 and not have them available in Gaffer N?

Reply:
> What did you think to the worry about it being annoying to save preferences in Gaffer N+1 and not have them available in Gaffer N?

It seems to match the preferences workflow of every other DCC, so people should be used to that. Personally, for something like houdini, I clean up (`rm`) those older folders regularly (usually around N-2)